migrations/create-patient-contacts.js

```diff
@@ -0,0 +1,177 @@
+var _ = require('underscore'),
+    db = require('../db'),
+    async = require('async');
+
+var BATCH_SIZE = 100;
+
+var filterThoseWithExistingContacts = function(batch, callback) {
+  db.medic.view('medic', 'patient_by_patient_shortcode_id', {
+      keys: batch.map(function(row) {
+        return row[0];
+      })
+    }, function(err, results) {
+      if (err) {
+        return callback(err);
+      }
+
+      var existingContactShortcodes = _.pluck(results.rows, 'key');
+
+      callback(batch.filter(function(row) {
+        return !_.contains(existingContactShortcodes, row[0]);
+      }));
+    });
+};
+
+var batchCreatePatientContacts = function(batch, callback) {
+  process.stdout.write('Of ' + batch.length + ' potential patients ');
```

Comment:
It's a slowly building sentence. This is how you do it to keep it on one line: console.log has a forced newline at the end.